src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
Maintaining this static list seems problematic in the long run. This has potential to go out of date as we introduce / remove errors and warnings. Is there some way to avoid that (can compiler tag such diagnostics somehow based on where they are generated from)?